app/questionnaire/value_source_resolver.py

```diff
@@ -183,4 +196,23 @@ def resolve(
             return self.response_metadata.get(value_source.get("identifier"))
 
         if source == "calculated_summary":
+            self.evaluate_completed_blocks_for_calculated_summary()
             return self._resolve_calculated_summary_value_source(value_source)
+
+    def evaluate_completed_blocks_for_calculated_summary(self) -> None:
```

Comment:
We can't do it like this as it would be very inefficient and would bottleneck eQ. This is evaluating the path for every section in the progress store. If calculated summary was the 10th section, this would evaluate the path for at least 9 + no. of repeating sections. If you have 4 repeats, that is 13 routing paths even if only 1 section is all you need.

Value source evaluator should be passed the block ids it needs to consider. It shouldn't have to work it out. 

If VSR is used by placeholder, then get placeholder should pass the block_ids, if used by RuleEvaluator, then RE should pass it. Or if directly using VSR then, it should pass it directly. You need to create a map of sections that the calculated summary depends on  instead of evaluating every section.

Reply:
Refactored now 👍